test/TileCacheTests.cpp

```diff
@@ -450,13 +450,14 @@ void TileCacheTests::testDisconnectMultiView()
     constexpr size_t repeat = 2;
     for (size_t j = 1; j <= repeat; ++j)
     {
+        TST_LOG("disconnectMultiView try #" << j);
+
+        // Make sure previous sessions have closed
+        kit_pid_helpers::waitForSpareKit(testname);
```

Comment:
Superficially this seems not to do what the previous test did surely? surely the previous code waited until there were no more document processes left open (?)

Reply:
It does the same thing.

countCoolKitProcesses is poorly named because it also waits, in this case for exactly one process.

waitForSpareKit waits until the spare kit is ready but also that there are no doc kits. So exactly one process, but this time it's definitely a spare. Is waitForSpareKit poorly named too? Perhaps. It's mostly used during setup, where you do want to make sure there are no old doc kits hanging around but usually there aren't and you end up waiting for the spare. I had considered a bunch of different names but didn't like any of them and settled on this. I'll think on it some more and see if I can come up with something clearer.

Reply:
> It does the same thing.

It is now really quite a large patch. You are 100% certain of that, presumably or you wouldn't state this so assertively and affirmatively I assume.

Somewhere along the line here - I think we've dropped a conceptual ball. The 'spare' kit stuff dates back from when we couldn't tell if a process was 'spare' or not - because we saw only the contents of /proc/<pid> So lots of the logic exists to remove the tiresome spare processes from our accounting.

Roll on to today - there is no need for this at all. We should not be waiting for situations where there are 0 spare kits eg. that looks like a nonsense; we should just not be looking at 'spare'-ness at all.

I expected a patch something like this:

diff --git a/test/countcoolkits.hpp b/test/countcoolkits.hpp
index b9c0435308..182cebfc2e 100644
--- a/test/countcoolkits.hpp
+++ b/test/countcoolkits.hpp
@@ -93,7 +93,7 @@ static int countCoolKitProcesses(const int expected,
 // (in the non-ideal event that) it's not 1,
 // it will cause testNoExtraCoolKitsLeft to
 // wait unnecessarily and fail.
-static int InitialCoolKitCount = 1;
+static int InitialCoolKitCount = 0;
 static std::chrono::steady_clock::time_point TestStartTime;
 
 static void testCountHowManyCoolkits()
diff --git a/test/test.cpp b/test/test.cpp
index 59fa4fafbc..0e24685442 100644
--- a/test/test.cpp
+++ b/test/test.cpp
@@ -321,7 +321,7 @@ std::set<pid_t> getForKitPids()
 /// How many live coolkit processes do we have ?
 int getCoolKitProcessCount()
 {
-    return getKitPids().size();
+    return getDocKitPids().size();
 }
 #endif
 

Can you test that instead and see where it falls down; perhaps it does not - perhaps it does.

Reply:
Quite possibly there is a test or two that really needs to know about spare kits; but the majority don't I expect will need touching.